### ISSUE TITLE
设置代理后经常超时问题。

### ISSUE THREAD

**Issue by thomas-cao** [cfug/dio#1624]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  iOS                        |      |
| Dio Version                    4.0.4                               |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps
在使用代理时经常性的导致接口超时，或者不响应。请问是什么问题导致的？

<img width="783" alt="image" src="https://user-images.githubusercontent.com/15570738/215304087-071ed340-feb0-4f6b-a6e3-0791126f6581.png">
